OpenTrons/opentrons-api
# feat(opentrons-ai-client): add auth0 package

# Overview
add auth0 package and useAuth0 hook to check the status of authentication and add `logout` button (this is for dev)
use getAccessTokenSilently to get access token from Auth0.

https://auth0.com/docs/get-started/applications/confidential-and-public-applications/user-consent-and-third-party-applications

close AUTH-204 AUTH-396


# Test Plan
1. `make -C opentrons-ai-client`
2. go to localhost:5173
3. the browser will be redirected to auth0 login/sign up page
4. log in (if you don't have an account, please sign up)
5. check the chat ui is shown
6. fill out the textarea and click send button
Check the app will receive the output from the BE (this would take some time)
In terms of the response display issue, it will be solved by [this pr](https://github.com/Opentrons/opentrons/pull/15182).
Also I have [a refactoring PR](https://github.com/Opentrons/opentrons/pull/15193) to fix current component structure issue (data fetch and token retrieving are done in InputPrompt component) 


7. click logout button
Check when clicking the logout button, the browser will show the login page again


# Changelog



# Review requests



# Risk assessment
low
